lib/ProductOpener/APIProductWrite.pm

```diff
@@ -85,6 +86,20 @@ sub skip_protected_field ($product_ref, $field, $moderator = 0) {
 	return 0;
 }
 
+=head2 skip_protected_image($product_ref, $imagefield, $moderator = 0)
+
+=cut
+
+sub is_protected_image ($product_ref, $imagefield, $moderator = 0) {
+
+	my $selected_uploader = get_selected_image_uploader($product_ref, $imagefield);
+	if ((not $server_options{producers_platform}) and (not $moderator) and (defined $selected_uploader)) {
```

Comment:
There should always be an uploader defined for each selected image. In order to know if the image is protected, we need to check if this uploader is the same as the owner of the product, with the owner field.

e.g. try this query: https://world.openfoodfacts.org/api/v3/product/8076800195057?fields=images,owner

the "front_fr" image is image imgid 69, and imgid 69's uploader is "org-barilla-france-sa" which is the same as the owner.

So this function should be changed to compare the uploader to the owner (if it exists).

Reply:
You can get the owner from $product_ref->{owner}